### PR TITLE
Add OCR attention model

### PR DESCRIPTION
**Pull request by qingqing01**
1.  Beam-search预测静态图可以跑通，动态图依赖： https://github.com/PaddlePaddle/Paddle/pull/22920 和 https://github.com/PaddlePaddle/Paddle/pull/23664

2. Add ImageFolder

see https://github.com/qingqing01/hapi/tree/ocr/examples/ocr

### CODE REVIEW

File: examples/ocr/data.py [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+CACHE_DIR_NAME = "attention_data"
+SAVED_FILE_NAME = "data.tar.gz"
+DATA_DIR_NAME = "data"
+TRAIN_DATA_DIR_NAME = "train_images"
+TEST_DATA_DIR_NAME = "test_images"
+TRAIN_LIST_FILE_NAME = "train.list"
+TEST_LIST_FILE_NAME = "test.list"
+
+
+class BatchCompose(object):
```
Comment (heavengate) [inline examples/ocr/data.py:32]: BatchCompose和Compose在vision的transform里有了

File: examples/ocr/data.py [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+            # label_out
+            samples[i].append(np.array(pad_label + [self.EOS], dtype='int64'))
+            mask = np.zeros((max_len + 1)).astype('float32')
+            mask[:len(label) + 1] = 1.0
+            # mask
+            samples[i].append(np.array(mask, dtype='float32'))
+        return samples
+
+
+class MyBatchSampler(fluid.io.BatchSampler):
```
Comment (heavengate) [inline examples/ocr/data.py:134]: MyBatchSampler -> OCRBatchSampler?

File: examples/ocr/data.py [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+        self.anno_file = anno_file
+        self._sample_infos = []
+        with open(anno_file, 'r') as f:
+            for i, line in enumerate(f):
+                w, h, im_name, labels = line.strip().split(' ')
+                h, w = int(h), int(w)
+                labels = [int(c) for c in labels.split(',')]
+                self._sample_infos.append(SampleInfo(i, h, w, im_name, labels))
+        #self._sample_infos = sorted(self._sample_infos,
+        #    key=lambda x: x.w)
```
Comment (heavengate) [inline examples/ocr/data.py:216]: delete?

File: examples/ocr/predict.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
+# Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
```
Comment (heavengate) [inline examples/ocr/predict.py:1]: 2018 -> 2020

File: examples/ocr/data.py (hunk 1 above)
Comment (qingqing01) [reply examples/ocr/data.py]: Done

File: examples/ocr/data.py (hunk 2 above)
Comment (qingqing01) [reply examples/ocr/data.py]: Done

File: examples/ocr/predict.py (hunk 4 above)
Comment (qingqing01) [reply examples/ocr/predict.py]: Done

File: examples/ocr/data.py (hunk 3 above)
Comment (qingqing01) [reply examples/ocr/data.py]: Done

File: examples/ocr/utility.py [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+        self.count = 0.
+
+    def accumulate(self):
+        return float(self.total) / self.count
+
+    def name(self):
+        return self._name
+
+
+class MyProgBarLogger(ProgBarLogger):
```
Comment (LielinJiang) [inline examples/ocr/utility.py:105]: It might be better to change MyProgBarLogger to OcrProgBarLogger

Comment (qingqing01) [reply examples/ocr/utility.py]: Change MyProgBarLogger to LoggerCallBack

**Pull request merged**